test/e2e/operator_test.go

```diff
@@ -1,26 +1,316 @@
+			err := runQuery(t, kubeClient, query, name, pods.Items[0].Name, url, bearerToken, true)
+			if err != nil {
+				return false, fmt.Errorf("Waiting for prometheus: %v", err)
+			} else {
+				return true, nil
+			}
+		})
+	})
+}
+
+func pdbCreate(t *testing.T, client *policyclientv1beta1.PolicyV1beta1Client, name string, labels map[string]string) error {
+	minAvailable := intstr.FromInt(1)
+	pdb := &policyv1beta1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: &minAvailable,
+			Selector:     &metav1.LabelSelector{MatchLabels: labels},
+		},
+	}
+	_, err := client.PodDisruptionBudgets(name).Create(pdb)
+	return err
+}
+
+func podCreate(t *testing.T, client *kubernetes.Clientset, name string, labels map[string]string) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: name,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{Name: "test", Image: "centos:7", Command: []string{"sh", "-c", "trap exit TERM; while true; do sleep 5; done"}},
+			}},
+	}
+	_, err := client.CoreV1().Pods(name).Create(pod)
+	return err
+}
+
+func locatePrometheus(t *testing.T, client *kubernetes.Clientset) (url, bearerToken string, ok bool) {
+	_, err := client.CoreV1().Services("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if kapierrs.IsNotFound(err) {
+		return "", "", false
+	}
+
+	for i := 0; i < 30; i++ {
+		secrets, err := client.CoreV1().Secrets("openshift-monitoring").List(metav1.ListOptions{})
+		if err != nil {
+			return "", "", false
+		}
+		for _, secret := range secrets.Items {
+			if secret.Type != corev1.SecretTypeServiceAccountToken {
+				continue
+			}
+			if !strings.HasPrefix(secret.Name, "prometheus-") {
+				continue
+			}
+			bearerToken = string(secret.Data[corev1.ServiceAccountTokenKey])
+			break
+		}
+		if len(bearerToken) == 0 {
+			time.Sleep(time.Second)
+			continue
+		}
+	}
+	if len(bearerToken) == 0 {
+		return "", "", false
+	}
+
+	return "https://prometheus-k8s.openshift-monitoring.svc:9091", bearerToken, true
+}
+
+const waitForPrometheusStartSeconds = 240
+
+type prometheusResponse struct {
+	Status string                 `json:"status"`
+	Data   prometheusResponseData `json:"data"`
+}
+
+type prometheusResponseData struct {
+	ResultType string       `json:"resultType"`
+	Result     model.Vector `json:"result"`
+}
+
+func getBearerTokenURLViaPod(t *testing.T, kclient *kubernetes.Clientset, ns, execPodName, url, bearer string) ([]byte, error) {
+	//cmd := fmt.Sprintf("curl -s -k -H 'Authorization: Bearer %s' %q", bearer, url)
+	// Retrieve the cert and key PEM of the current CA via kube-controller-manager-operator serving-cert secret
+	secret, err := kclient.CoreV1().Secrets(kcmoNamespace).Get(kcmoSigningKeySecret, metav1.GetOptions{})
```

Comment:
That configuring tls with the serving cert works is an accident of implementation. The signing CA is included with the serving cert, allowing it to be reused as a ca bundle. However, this is not good practice and at some point in the future serving certs will not include the signing CA. Please source an actual ca bundle from the service ca operator instead as per the following example (create annotated configmap, wait ca bundle to be added to configmap):

https://github.com/openshift/service-ca-operator/blob/ca-validity-duration/test/e2e/e2e_test.go#L329

Reply:
ok, got it.. working on it...

Reply:
done, i'm obtaining the caBundle